sdds/reader.py

```diff
@@ -5,36 +5,84 @@
 
 import numpy as np
 
-from sdds.classes import (SddsFile, Column, Parameter, Definition, Array, Data, Description,
-                          ENCODING, NUMTYPES_CAST, NUMTYPES_SIZES, get_dtype_str)
+from sdds.classes import (ENCODING, NUMTYPES_CAST, NUMTYPES_SIZES, Array,
+                          Column, Data, Definition, Description, Parameter,
+                          SddsFile, get_dtype_str)
 
+# ----- Providing Opener Abstractions for the Reader ----- #
 
-def read_sdds(file_path: Union[pathlib.Path, str], endianness: str = None) -> SddsFile:
+OpenerType = Callable[[os.PathLike], AbstractContextManager[io.BufferedIOBase]]
+
+binary_open = partial(open, mode="rb")  # default opening mode, simple sdds files
+gzip_open = partial(gzip.open, mode="rb")  # for gzip-compressed sdds files
+
+
+# ----- Reader Function ----- #
+
+def read_sdds(file_path: Union[pathlib.Path, str], endianness: str = None, opener: OpenerType = binary_open) -> SddsFile:
     """
-    Reads SDDS file from the specified ``file_path``.
+    Reads an SDDS file from the specified ``file_path``.
 
     Args:
         file_path (Union[pathlib.Path, str]): `Path` object to the input SDDS file. Can be a
             `string`, in which case it will be cast to a `Path` object.
-        endianness (str): Endianness of the file. Either 'big' or 'little'.
-                          If not given, the endianness is either extracted from
-                          the comments in the header of the file (if present)
-                          or determined by the machine you are running on.
-                          Binary files written by this package are all big-endian,
-                          and contain a comment in the file.
+        endianness (str): Endianness of the file, either 'big' or 'little'. If not given,
+            the endianness is either extracted from the comments in the header of the file
+            (if present) or determined by the machine you are running on. Binary files
+            written by this package are all big-endian, and contain a comment in the file.
+        opener (OpenerType): Callable to open the SDDS file. Uses `open(file, mode="rb")` by
+            default. One can use provided openers for specific format or bring their own, see
+            the examples below.
 
     Returns:
         An `SddsFile` object containing the loaded data.
+    
+    Examples:
+
+        To read a typical file, one can use the default options:
+
+        .. code-block:: python
+
+            import sdds
+
+            data = sdds.read("some/location/to/file.sdds")
+
+
+        To read a ``gzip``-compressed file, use the provided opener function:
+
+        .. code-block:: python
+
+            import sdds
+            from sdds.reader import gzip_open
+
+            data = sdds.read("some/location/to/file.sdds.gz", opener=gzip_open)
+
+        To read another specific compression format, bring your own opener abstraction:
```

Comment:
... which takes a single parameter, which will be the path-like object pointing to the file.

Reply:
an returns a context-manager, providing the byte-data of the file.

(or something)